src/Nethermind/Nethermind.Api/IEngineRequestsTracker.cs

```diff
@@ -0,0 +1,10 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Api;
+
+public interface IEngineRequestsTracker
+{
+    void OnForkchoiceUpdatedCalled();
```

Comment:
I think that the introduction of this tracker is not needed. Its methods are called in the `EngineRpcModule` which is constructed in each plugin according to its needs. For `Optimism`, one could introduce a decorator for the original components like `IForkchoiceUpdatedHandler` and pass it down to the `EngineRpcModule`. This would keep the original shape of the `EngineRpcModule`. The seam seems to be ready to be extended without introduction of additional steps.